로그인 구현

kihyuk-jeong opened the pull request:
- 로그인 구현 : 인터셉터 + 어노테이션 방식으로 구현
- 로그인 관련 테스트코드 작성

Hosick committed:
[#1]프로젝트 설정

-패키지명 변경
-profile 구분

kihyuk-jeong committed:
etc : application.yml 설정 추가 및 주석 추가

1. profile 별로 ddl-auto 속성 설정
2. 각 설정에 대한 주석 추가

[#1] 프로젝트 설정

Hosick committed:
etc: applicaition.yml 수정

-오타 수정
-주석 수정

[#1] 프로젝트 설정

kihyuk-jeong committed:
feature : 로그인/로그아웃 기능 추가 - 1차

<description>
1. 세션을 이용한 로그인/로그아웃 기능 구현
2. 관련 테스트코드 구현

<issue-number>
 #6 : 회원가입 구현

Hosick committed:
[#1] application.yml 분리

kihyuk-jeong committed:
[#6] etc : 테스트코드 추가

<description>
1. 테스트코드 추가

<issue-number>
 #6

Hosick committed:
[#1]chore: yml 분리

- 모든 profile에서 사용되는 전역설정 분리

kihyuk-jeong committed:
[#6] 로그인 구현

[#6] refactor : 회원가입 비즈니스 로직과 서비스 비즈니스 로직 분리

<issue-number>
 #6 로그인 구현

Hosick committed:
[#1]chore: 리뷰 반영

-logging 옵션 분리 적용
-production profile 용 yml 생성

kihyuk-jeong committed:
[#6] feature : 로그인 확인을 위한 interceptor 추가

<description>
1. 로그인 확인을 위한 인터셉터 추가

<issue-number>
 #6

kihyuk-jeong committed:
[#6] etc : 로그인 체크 수정 / 테스트 코드 추가

1. 어노테이션을 이용한 로그인 체크 구현
2. 로그인관련 테스트코드 추가
3. 로직 변경으로 인한 테스트코드 수정
4. goole checkstyle 일괄 적용

<issue-number>
 #6

kihyuk-jeong committed:
[#6] feature : 로그인 인터셉터 구현

<description>
<issue-number>
 #6

f-lab-dev commented on src/main/java/com/flab/shoeauction/common/utils/encrytion/EncryptionUtils.java:
```diff
@@ -0,0 +1,7 @@
+package com.flab.shoeauction.common.utils.encrytion;
+
+public interface EncryptionUtils {
```
암호화 관련 비지니스 로직을 구현하므로 `Utils`보단 `Service`가 낫지 않을까요?

kihyuk-jeong replied:
공통적으로 사용하는 부분이라고 생각돼서 `utils`로 네이밍 했는데 생각해 보니 로그인과 회원가입, 회원정보 변경 등 `User`관련 비즈니스 로직인 것 같습니다. `Service`로 수정하는 것이 나을 것 같습니다.!

f-lab-dev commented on src/main/java/com/flab/shoeauction/common/utils/encrytion/SHA256Encryption.java:
```diff
@@ -0,0 +1,25 @@
+import org.springframework.stereotype.Component;
+
+@Component
+public class SHA256Encryption implements EncryptionUtils {
```
이거는 회원가입에 필요한게 아닐까요? 왜 로그인에 섞여있는지 궁금합니다~

kihyuk-jeong replied:
암호화 알고리즘을 추상화하는 방식으로 수정하는 과정에서 잘못 커밋한 것 같습니다.

f-lab-dev commented on src/main/java/com/flab/shoeauction/user/controller/UserController.java:
```diff
@@ -60,4 +61,20 @@ public ResponseEntity signUp(@RequestBody @Valid UserDto signUpDto) {
+    @PostMapping("/login")
+    public ResponseEntity login(@RequestBody LoginDto loginDto) {
+        String email = loginDto.getEmail();
+        String password = encryptionUtils.encrypt(loginDto.getPassword());
```
`암호화한다`는 것은 서비스의 책임일까요? 컨트롤러의 책임일까요?
각 레이어별 역할에 대해 알아보시면 좋을 것 같네요~

kihyuk-jeong replied:
컨트롤러가 `암호화`의 기능까지 책임지고 있다는 건 SRP 원칙에 어긋나는 것 같습니다. 서비스 로직으로 분류하겠습니다.

f-lab-dev commented on src/test/java/com/flab/shoeauction/user/controller/UserControllerTest.java:
```diff
@@ -33,69 +32,89 @@
 
 
+    @Test
+    @DisplayName("로그인 성공")
```
단순 성공이 아니라 `어떨때 로그인이 성공한다` 라는 명확한 조건을 명시해주면 좋을 것 같습니다

kihyuk-jeong replied:
이메일과 비밀번호가 일치할때 로그인이 성공한다는 조건 추가하고
모든 테스트코드마다 명확하게 조건을 명시하도록 구현하겠습니다~

kihyuk-jeong committed:
[#6] refactor : 리뷰반영

<description>
1. EncryptionUtils -> EncryptionService
2. 컨트롤러에서 처리하던 loginDto 암호화 작업 서비스로 분류
 + (로직 변경에 따른 테스트코드 수정)
3. 테스트 코드 명확성 부여

<issue-number>
    #6 : 회원가입 구현

f-lab-dev commented on src/main/java/com/flab/shoeauction/user/service/LoginService.java:
```diff
@@ -13,15 +15,22 @@
     }
 
+    private String passwordEncryption(String password) {
+        return encryptionService.encrypt(password);
```
이거는 단순하게 `encryptionService.encrypt(password);`를 감싸기만 하는데 이 메소드를 굳이 선언할 필요가 있을까요?

kihyuk-jeong replied:
가독성을 위해서 메서드로 선언했는데 단순하게 감싸기만 한다면 굳이 그럴 필요가 없는 것 같습니다. 추가로 회원가입 시 `UserDto`클래스에서 암호화를 처리하는 방법과 동일하게 `LoginDto` 클래스에서 암호화를 처리하도록 로직을 일관성 있게 수정했습니다.!

f-lab-dev commented on src/main/java/com/flab/shoeauction/user/service/encrytion/SHA256Encryption.java:
```diff
@@ -1,10 +1,10 @@
-@Component
-public class SHA256Encryption implements EncryptionUtils {
+@Service
+public class SHA256Encryption implements EncryptionService {
```
이건 취향차이긴 한데 여기도 부모 클래스 따라서 이름에 `Service`를 붙여주는게 낫지 않을까요?

kihyuk-jeong replied:
이 부분은 `XXXServiceImpl`처럼 service를 구현했다는 의미 대신 SHA256이라는 네임을 통해 충분히 의미가 전달될 수 있을 것 같아서 이렇게 처리했는데 `@Service`를 사용하고 있는 만큼 Service라는 이름을 붙여서 의미를 더 명확하게 전달할 수 있을 것 같습니다. 반영하겠습니다.!

kihyuk-jeong committed:
[#6] feature : 현재 로그인된 유저 정보 로직 추가 및 리뷰반영

[1. 리뷰반영]
1-1. SHA256Encryption의 이름을 SHA256EncryptionService 수정함으로써 service 비즈니스 로직이라는 명확한 의미 부여
1-2. 불필요한 메소드 제거 및 로그인시 사용되는 암호화 로직 LoginDto 클래스로 이동
[2. 현재 로그인된 유저 정보를 불러오는 마이페이지(profile) 기능 구현]
2-1. 커스텀 어노테이션을 이용한 HandlerMethodArgumentResolver 구현
2-2. 마이페이지 관련 테스트코드 작성 및 로그인/로그아웃 테스트코드 수정

<issue-number>
 #6 : 로그인 구현
 #13 : 로그인 유저 정보 요청 기능 구현(마이페이지)

f-lab-dev commented on src/main/java/com/flab/shoeauction/common/configs/LoginUserArgumentResolver.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
+        NativeWebRequest webRequest, WebDataBinderFactory binderFactory) throws Exception {
+        return loginService.getLoinUser();
```
`getLoinUser` 오타로 보이네요~

f-lab-dev commented on src/main/java/com/flab/shoeauction/user/controller/UserController.java:
```diff
@@ -73,4 +74,10 @@ public ResponseEntity logout() {
     }
+
+    @LoginCheck
+    @GetMapping("/profile")
```
이런거는 내 정보를 가져오는거니까 `GET /my-infos` 이런식으로 URL을 설계해보면 어떨까요?

kihyuk-jeong replied:
URL 설계 부분에서도 고민을 많이 해보겠습니다. 반영하겠습니다 감사합니다!

kihyuk-jeong committed:
Merge branch 'settings' into feature/3/kh

kihyuk-jeong committed:
[#6] refactor : 리뷰반영

1. 내 정보를 보여주는 URL 명칭 my-info로 변경
2. 오타수정

<issue-number>
 #6

kihyuk-jeong committed:
[#6] feature : sessio 불일치 해결을 위한 redis 추가 외 1

1. redis 추가
2. my-info (내 정보) 단위테스트 추가

f-lab-dev commented on src/main/java/com/flab/shoeauction/user/controller/UserController.java:
```diff
@@ -76,8 +76,9 @@ public ResponseEntity logout() {
-    @GetMapping("/profile")
-    public ResponseEntity<UserInfoDto> myPage(@LoginUser UserInfoDto userInfoDto) {
-        return ResponseEntity.ok(userInfoDto);
+    @GetMapping("/my-info")
```
URL은 단수형이 아니라 복수형이여야합니다. 이전에 전달드린 API 문서를 확인해보시면 좋겠네요~

kihyuk-jeong replied:
네. 참고해서 수정하겠습니다.!

kihyuk-jeong committed:
refactor : 로그인 / 세션 불일치 문제 브랜치 분리 및 아키텍쳐 구조 수정

 1. 로그인 구현과 세션 불일치 문제를 해결하기 위한 세션 스토리지 추가 이슈 분리
 2. 팀원과 아키텍쳐 구조 통일
 3. HttpStatus 상수 이름 변경
 4. 리뷰반영(my-info - > my-infos : URL은 복수여야 한다.)

kihyuk-jeong commented:
2021-01-29 변경사항 : 머지를 위해 @Hosick 님과 아키텍처 구조를 통일하였습니다.

f-lab-dev commented on src/main/java/com/flab/shoeauction/common/configs/WebConfig.java:
```diff
@@ -1,5 +1,9 @@
 
+/**
+ *
+ */
```
이런 의미없는 코드는 커밋하지 않는게 좋습니다~ 커밋하기전에 변경내용을 확인하고 push 해주세요~

kihyuk-jeong replied:
구현한 코드에 대한 설명을 넣으려다가 실수로 작성한 것 같습니다. 커밋 전에 항상 확인하는 습관 들이겠습니다.!

kihyuk-jeong committed:
refactor : 리뷰 반영

불필요한 주석 삭제
